lib/mongo/monitoring/event/secure.rb

```diff
@@ -53,8 +54,14 @@ module Secure
         #
         # @since 2.1.0
         def redacted(command_name, document)
-          if REDACTED_COMMANDS.include?(command_name.to_s) &&
-            !%w(1 true yes).include?(ENV['MONGO_RUBY_DRIVER_UNREDACT_EVENTS']&.downcase)
+          if %w(1 true yes).include?(ENV['MONGO_RUBY_DRIVER_UNREDACT_EVENTS']&.downcase)
+            return document
+          end
+
+          if REDACTED_COMMANDS.include?(command_name.to_s)
+            BSON::Document.new
+          elsif %w(hello ismaster isMaster).include?(command_name.to_s) &&
+            !!document['speculativeAuthenticate']
```

Comment:
If the spec requires omitting the entire event payload rather than just the `speculativeAuthenticate` subkey, as I think it does, can that please be added as a note here for future readers?

Reply:
As I understood the spec, we must omit the entire payload. A comment added.